Source/loadsave.cpp

```diff
@@ -284,17 +284,17 @@ static void LoadPlayer(LoadHelper *file, int p)
 {
 	PlayerStruct *pPlayer = &plr[p];
 
-	pPlayer->_pmode = file->nextLE<Sint32>();
+	pPlayer->_pmode = (PLR_MODE)file->nextLE<Sint32>();

```

Comment:
Doesn't this work?
```c++
nextLE<PLR_MODE>
```

Reply:
It's not safe as we need the data to be read from the file in fixed sizes no matter how enums may be implemented on various systems.

Reply:
> An enum is only guaranteed to be large enough to hold int values. The compiler is free to choose the actual type used based on the enumeration constants defined so it can choose a smaller type if it can represent the values you define.

Reply:
Ah got it. I got confused for a moment since C# enums allow you to explicitly define the underlying type. Seems to only be available in C++11 and newer though.

Reply:
well, we are on C++11 so :shrug:

Reply:
oh! then we should be able to use that XD

https://stackoverflow.com/a/1122109/1946412

Reply:
yep, but I would rather not have the old savegame format dictate what types we use :)